### PR TITLE
Dynamic Simulation - Rework on curve configuration to use expert filter

### PR DESCRIPTION
**Pull request by thangqp**
This PR hasn't any functional evolution on curve configuration. The GUI is unchanged

Related PRs:

Back-end :
https://github.com/gridsuite/study-server/pull/476
https://github.com/gridsuite/network-map-server/pull/147
https://github.com/gridsuite/filter-server/pull/78

Demo

curve_IEEE14

### CODE REVIEW

File: src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.js [hunk 1]
```diff
@@ -46,253 +47,172 @@ const styles = {
...
-                    // get countries codes
-                    let countries = [
-                        ...vals.flat().reduce((set, substation) => {
-                            substation.countryCode &&
-                                set.add(substation.countryCode);
-                            return set;
-                        }, new Set()),
-                    ];
-
+            fetchAllCountries(studyUuid, currentNode.id)
```
Comment (SlimaneAmar) [inline src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.js:110]: When node is invalid we got en error

Comment (thangqp) [reply src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.js:110]: Done : https://github.com/gridsuite/gridstudy-app/commit/fef4bab75d83dc7ef8bd57d92ba9343d123b6c51#diff-004e2fd93b0e13e660ba95c75ebe656a464ba5f4810c28fd61ce1feb353ceaf6R24

File: src/components/dialogs/filter/expert/expert-filter.type.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    combinator: CombinatorType;
+    dataType: DataType;
+    rules: (RuleTypeExport | RuleGroupTypeExport)[];
+}
+
+export interface ExpertFilter {
+    id?: UUID;
+    type: 'EXPERT';
+    equipmentType: string; // TODO must be EquipmentType enum
+    rules: RuleGroupTypeExport;
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/components/dialogs/filter/expert/expert-filter.type.ts]: Done : https://github.com/gridsuite/gridstudy-app/pull/1658/commits/bbc70648207adb51ff44cbd7a1bf219b5cc9ba82

File: src/services/study/filter.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { backendFetchJson } from '../utils';
+import { UUID } from 'crypto';
+import { getStudyUrlWithNodeUuid } from './index';
+import { ExpertFilter } from '../../components/dialogs/filter/expert/expert-filter.type';
+
+export function evaluateFilter(
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/services/study/filter.ts]: Done: https://github.com/gridsuite/gridstudy-app/pull/1658/commits/f0e19882524a7ec5b7181baba57371ee5ac532ef

**Review by SlimaneAmar — APPROVED** [review]

**Review by EstherDarkish — APPROVED** [review]
Code OK Tests OK

**Pull request merged**